### PR TITLE
remove some redundant variables and type params

### PR DESCRIPTION
**Pull request by SimonCropp**
u may want to squash this one since i decided to revert the sum commit

### CODE REVIEW

File: tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs [hunk 1]
```diff
@@ -150,7 +150,7 @@ public void NonMutatingEncodePreservesPaletteCount()
                 outStream.Position = 0;
                 var clone = Image.Load<Rgba32>(outStream);
 
-                GifMetadata cloneMetaData = clone.Metadata.GetFormatMetadata<GifMetadata>(GifFormat.Instance);
+                GifMetadata cloneMetaData = clone.Metadata.GetFormatMetadata(GifFormat.Instance);
```
Comment (dlemstra) [inline tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: Maybe better to use `var cloneMetaData` here instead?

Comment (SimonCropp) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: do u want another pr that makes the use of `var` consistent?

Comment (dlemstra) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: That would be nice, but you can probably sneak this one in?

Comment (antonfirsov) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: I think, we should merge it as-is now to keep the ball rolling.

**Comment by JimBobSquarePants** [comment]
I’m gonna merge this. I expect a lot of test code cleanup over the next few months anyway.

File: tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs (hunk 1 above)
Comment (SimonCropp) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: @dlemstra @antonfirsov @JimBobSquarePants so re var. whats your preferences? "just var everywhere"?

Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: The ide should hint via editorconfig. Var only when the type is apparent.

Comment (SimonCropp) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: weird. i dont get a hint?

Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: Very weird, what’s your dev setup? Any plugins etc?

Comment (SimonCropp) [reply tests/ImageSharp.Tests/Formats/Gif/GifEncoderTests.cs]: I will troubleshoot when i get home, and get back to you

**Pull request merged**